### PR TITLE
docs: Docs for sensitive variables

### PR DESCRIPTION
**Pull request by pselle**
This adds documentation for the `sensitive` argument for variable blocks. This PR also re-organizes the variable page to have an index of the argument sections, to make the page more navigable.

### CODE REVIEW

File: website/docs/configuration/variables.html.md [hunk 1]
```diff
@@ -76,51 +76,35 @@ assign a value to the variable from outside and to reference the variable's
...
-
-- `source`
-- `version`
-- `providers`
-- `count`
-- `for_each`
-- `lifecycle`
-- `depends_on`
-- `locals`
+_except_ the following: `source`, `version`, `providers`, `count`, `for_each`, `lifecycle`, `depends_on`, `locals`.
```
Comment (mildwonkey) [inline website/docs/configuration/variables.html.md:79]: I'm curious about this change;  these are in-page links on the website (I believe that's what the extra `-` does): 

https://www.terraform.io/docs/configuration/variables.html#declaring-an-input-variable

File: website/docs/configuration/variables.html.md [hunk 2]
```diff
@@ -220,6 +208,91 @@ that includes the sentences given in `error_message`. The error message string
 should be at least one full sentence explaining the constraint that failed,
 using a sentence structure similar to the above examples.
 
+### Sensitive
+
+[inpage-sensitive]: #sensitive
+
+-> This feature was introduced in Terraform CLI v0.14.0.
+
+The `sensitive` argument on a variable block is a boolean value that, when provided, limits the output of the Terraform `plan` or `apply` when that variable is used. A provider can define [an attribute as sensitive](/docs/extend/best-practices/sensitive-state.html#using-the-sensitive-flag), which prevents the value of that attribute from being displayed in logs or regular output. The `sensitive` argument on variables allows users to replicate this behavior for values in their configuration, by defining a variable as `sensitive`.
```
Comment (mildwonkey) [inline website/docs/configuration/variables.html.md:217]: I might rephrase this sentence a bit, inspired by our sensitive outputs documentation (https://www.terraform.io/docs/configuration/outputs.html):

Setting a variable as `sensitive` argument prevents Terraform from showing its value in the `plan` and `apply` output.

File: website/docs/configuration/variables.html.md [hunk 3]
```diff
@@ -220,6 +208,91 @@ that includes the sentences given in `error_message`. The error message string
...
+    }
+
+Plan: 1 to add, 0 to change, 0 to destroy.
+```
+
+#### Cases where Terraform may disclose a sensitive variable
+
+Variable values marked as sensitive will display in state. Much like provider-side sensitive values, designating a value as sensitive only limits its display in logs or output, not in state.
+
+Similarly, `sensitive` argument does not have an impact in other Terraform commands such as `console` or `show`. Those commands are meant to ex as this argument is intended to reduce exposure of data in, for example, external logs or aggregation.
```
Comment (mildwonkey) [inline website/docs/configuration/variables.html.md:254]: > Those commands are meant to ex as this 
Not quite sure what is intended here?

File: website/docs/configuration/variables.html.md [hunk 4]
```diff
@@ -220,6 +208,91 @@ that includes the sentences given in `error_message`. The error message string
...
+      + name    = (sensitive)
+      + address = (sensitive)
+    }
+
+Plan: 1 to add, 0 to change, 0 to destroy.
+```
+
+#### Cases where Terraform may disclose a sensitive variable
+
+Variable values marked as sensitive will display in state. Much like provider-side sensitive values, designating a value as sensitive only limits its display in logs or output, not in state.
```
Comment (mildwonkey) [inline website/docs/configuration/variables.html.md:252]: Is this saying they are stored in state (which I know they are) or that they will be displayed in `state show`, or both? "display in state" might seem ambiguous to some readers.

File: website/docs/configuration/variables.html.md [hunk 5]
```diff
@@ -76,51 +76,35 @@ assign a value to the variable from outside and to reference the variable's
...
-in the following section.
+## Arguments
+
+Terraform CLI defines the following optional arguments for variable declarations:
+
+- [`default`][inpage-default] - A default value which then makes the variable optional.
+- [`type`][inpage-type] - This argument specifies what value types are accepted for the variable.
+- [`description`][inpage-description] - This specifies the input variable's documentation.
+- [`validation`][inpage-validation] - A block to define validation rules, usually in addition to type constraints.
+- [`sensitive`][inpage-sensitive] - A boolean value that can limit Terraform UI output when the variable is used in configuration, to avoid accidental disclosure in logs or other UI.
```
Comment (mildwonkey) [inline website/docs/configuration/variables.html.md:93]: Nitpick: `sensitive` is  an argument, not a value. I might say "An argument that can limit ..."

File: website/docs/configuration/variables.html.md [hunk 6]
```diff
@@ -220,6 +208,91 @@ that includes the sentences given in `error_message`. The error message string
...
+Plan: 1 to add, 0 to change, 0 to destroy.
+```
+
+#### Cases where Terraform may disclose a sensitive variable
+
+Variable values marked as sensitive will display in state. Much like provider-side sensitive values, designating a value as sensitive only limits its display in logs or output, not in state.
+
+Similarly, `sensitive` argument does not have an impact in other Terraform commands such as `console` or `show`. Those commands are meant to ex as this argument is intended to reduce exposure of data in, for example, external logs or aggregation.
+
+Defining sensitivity at the variable level is a configuration-centered concept, and values are sent to providers without any obfuscation; as such, a provider error could disclose a value depending on if the value is included in the error message. For example, if a provider returns an error with `"Invalid value 'foo' for field"`, the value will be displayed in output as a consequence of this error.
```
Comment (mildwonkey) [inline website/docs/configuration/variables.html.md:256]:
```suggestion
A ` sensitive` variable level is a configuration-centered concept, and values are sent to providers without any obfuscation. A provider error could disclose a value if that value is included in the error message. For example, a provider might return the following error even if "foo" is a sensitive value: `"Invalid value 'foo' for field"`
```

This paragraph could use a bit more of a direct and active tone though my suggestion could also use some work :)

File: website/docs/configuration/variables.html.md [hunk 7]
```diff
@@ -220,6 +208,91 @@ that includes the sentences given in `error_message`. The error message string
...
+      + prefix    = (sensitive)
+      + separator = "-"
+    }
+
+Plan: 1 to add, 0 to change, 0 to destroy.
+
+...
+
+random_pet.animal: Creating...
+random_pet.animal: Creation complete after 0s [id=jae-known-mongoose]
```
Comment (mildwonkey) [inline website/docs/configuration/variables.html.md:274]: This is a great example!

**Review by mildwonkey — CHANGES_REQUESTED** [review]
Overall this looks great! I left some thoughts; I'm not the best writer so please don't hesitate to disagree or make changes that _don't_ use my suggestions - the only question I'd say is blocking this is changing the bulleted list of valid identifiers (with links) to an inline list without links.

File: website/docs/configuration/variables.html.md (hunk 1 above)
Comment (pselle) [reply website/docs/configuration/variables.html.md:79]: It is solely to make them take up less real-estate on the page. They are links, but you'll note that they just link to themselves ... which adds no extra context. https://www.terraform.io/docs/configuration/variables.html#locals

File: website/docs/configuration/variables.html.md (hunk 3 above)
Comment (pselle) [reply website/docs/configuration/variables.html.md:254]: Oops, just a mistake. Thanks for flagging this!

File: website/docs/configuration/variables.html.md (hunk 4 above)
Comment (pselle) [reply website/docs/configuration/variables.html.md:252]: Both

File: website/docs/configuration/variables.html.md (hunk 5 above)
Comment (pselle) [reply website/docs/configuration/variables.html.md:93]: The "value" here refers to the right side of the argument. I'm worried that if I change it here, it doesn't make sense for ex. `default`

File: website/docs/configuration/variables.html.md (hunk 1 above)
Comment (mildwonkey) [reply website/docs/configuration/variables.html.md:79]: got it, thanks!

**Review by mildwonkey — APPROVED** [review]
I am sorry about the misguided question about the invalid identifiers!

**Pull request merged**